apps/web/src/components/tx-flow/flows/SafeAppsTx/ReviewSafeAppsTx.tsx

```diff
@@ -40,7 +41,8 @@ const ReviewSafeAppsTx = ({
     }
 
     createSafeTx().then(setSafeTx).catch(setSafeTxError)
-  }, [txs, setSafeTx, setSafeTxError, params])
+    setTxOrigin(getTxOrigin(app))
```

Comment:
I moved it into this useEffect but let me know if you can think of a better place to handle this logic of setting the tx origin. Previously it was on the Continue button callback together with the TxNote logic.

Reply:
I'm not sure what would happen, but isn't it better if you move the setTxOrigin call inside of the then part?

Right now even if the tx fails and is catched you would set the origin.

Reply:
Another alternative is that we pass the origin down to the `ReviewTransactionContent` and include it in the `onContinueClick` callback. This would match the previous behaviour but I like this approach more to handle it higher up since it only needs to be done for safe app transactions so no need to add this complexity to the review components.